.github/workflows/linux.yml

```diff
@@ -16,476 +16,490 @@ concurrency:
         with:
-          # Should save cache only if run in the master branch of the base repo
-          # github.ref_name is 'ref/PR_#' in case of the PR, and 'branch_name' when executed on push
-          save-always: ${{ github.ref_name == 'master' && 'true' || 'false'  }}
-          path: ${{ env.CCACHE_DIR }}
-          key: ${{ runner.os }}-${{ runner.arch }}-ccache-genai-release-${{ github.sha }}
-          restore-keys: |
-            ${{ runner.os }}-${{ runner.arch }}-ccache-genai-release
+          name: ov_samples_${{ matrix.build-type }}
+          path: ${{ env.INSTALL_DIR }}/*.tar.gz
+          if-no-files-found: 'error'
+          
+  genai_tests_archive:
+    name: Test ${{ matrix.test.name }} (archive/${{ matrix.build-type }})
+    needs: [ openvino_download, genai_build_cmake ]
+    strategy:
+      fail-fast: false
+      matrix:
+        build-type: [Release]
+        test:
+          - name: 'Chat Generate Api'
+            cmd: 'tests/python_tests/test_chat_generate_api.py -k "not Qwen2-0.5B-Instruct"'
+          - name: 'Whisper Generate API'
+            cmd: 'tests/python_tests/test_whisper_generate_api.py -k "${PRECOMMIT_FILTER} and not distil-whisper"'
+    timeout-minutes: 45
+    defaults:
+      run:
+        shell: bash
+    runs-on: aks-linux-4-cores-16gb
+    container:
+      image: openvinogithubactions.azurecr.io/ov_test/ubuntu_22_04_x64:${{ needs.openvino_download.outputs.docker_tag }}
+      volumes: 
+        - /mount:/mount
+        - ${{ github.workspace }}:${{ github.workspace }}
 
-      - name: Build genai
+    env:
+      INSTALL_DIR: ${{ github.workspace }}/install
+      SRC_DIR: ${{ github.workspace }}/src
+      BUILD_DIR: ${{ github.workspace }}/build
+      PRECOMMIT_FILTER: 'test_language or timestamps'
+        
+    steps:
+      - name: Clone openvino.genai
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          path: ${{ env.SRC_DIR }}
+          submodules: recursive
+          
+      - name: Download Build Artifacts
+        uses: actions/download-artifact@fa0a91b85d4f404e444e00e005971372dc801d16 # v4.1.8
+        with:
+          pattern: "{${{ needs.openvino_download.outputs.ov_artifact_name }},genai_archive_${{ matrix.build-type }}}"
+          path: ${{ env.INSTALL_DIR }}
+          merge-multiple: true
+      
+      - name: Extract Artifacts
+        run: pigz -dc ${{ env.GENAI_ARCHIVE_NAME }} | tar -xf - -C ${{ env.INSTALL_DIR }}
+        working-directory: ${{ env.INSTALL_DIR }}
+  
+      - name: Install Test Requirements
         run: |
-          source ${OV_INSTALL_DIR}/setupvars.sh
-          cmake -DCMAKE_BUILD_TYPE=Release -S ./ -B ./build/
-          cmake --build ./build/ --config Release --target py_openvino_genai -j
-
-      - name: Test bindings
+          python -m pip install -r ./tests/python_tests/requirements.txt
+          python -m pip uninstall openvino openvino-tokenizers -y
```

Comment:
I would uninstall GenAI as well
currently, it's not in this requirements.txt file, but could be in future..

Reply:
removed that at all